pipeline/detection.py

```diff
@@ -706,6 +757,121 @@ def extract_sources_sep(self, image):
+
+        Sources are detected as points in the score image that are
+        above the given threshold. If points are too close, they
+        will be merged into a single source.
+        The output SourceList will include the aperture and PSF photometry,
+        and the x,y positions will be given using the source centroid.
+
+        # TODO: should we do iterative PSF tapered centroiding?
+
+        Parameters
+        ----------
+        image: Image
+            The image to extract sources from. Must have a score, or a PSF,
+            that can be gotten from one of these places (in order):
+            - a PSF object loaded on the Image object itself.
+            - a zogy_psf attribute attached to the Image object.
+            - a PSF object of the new_image attribute of the Image object.
+            If the image also has a zogy_alpha attribute, it will be used
+            to extract the PSF photometry.
+
+        Returns
+        -------
+        sources: SourceList
+            The list of sources detected in the image.
+            This contains a table where each row represents
+            one source that was detected, along with all its properties.
+
+        """
+        score = image.score.copy()
+
+        if score is None:
+            raise NotImplementedError('Still need to add the matched-filter cross correlation! ')
+
+        psf_image = None
+        if image.psf is not None:
+            psf_image = image.psf.get_clip()
+        elif getattr(image, 'zogy_psf', None) is not None:
+            psf_image = image.zogy_psf
+
+        fwhm = image.fwhm_estimate
+        if fwhm is None:
+            fwhm = image.new_image.fwhm_estimate
+        if fwhm is None and image.psf is not None:
+            fwhm = image.psf.fwhm_pixels
+        if fwhm is None and image.new_image.psf is not None:
+            fwhm = image.new_image.psf.fwhm_pixels
+
+        if fwhm is None:
+            raise RuntimeError("Cannot find a FWHM estimate from the given image or its new_image attribute.")
+
+        # typical scale of PSF is X times the FWHM
+        fwhm_pixels = max(int(np.ceil(fwhm * self.pars.separation_fwhms / image.instrument_object.pixel_scale)), 1)
+        # remove flagged pixels
+        score[image.flags > 0] = np.nan
+
+        # remove the edges
+        border = fwhm_pixels * 2
+        score[:border, :] = np.nan
+        score[-border:, :] = np.nan
+        score[:, :border] = np.nan
+        score[:, -border:] = np.nan
+
+        # normalize the score based on sigma_clipping
+        # TODO: we should check if we still need this after b/g subtraction on the input images
+        mu, sigma = sigma_clipping(score)
+        score = (score - mu) / sigma
+        det_map = abs(score) > self.pars.threshold  # catch negative peaks too (can get rid of them later)
+
+        # dilate the map to merge nearby peaks
+        struct = np.ones((1 + 2 * fwhm_pixels, 1 + 2 * fwhm_pixels))
```

Comment:
Do we want to do a circular rather than square dilation?  More effort, of course (though only a couple of lines).  But, doing it as squares means that the confusion length scale is going to be ~√2 different between 0° and 45°

(And, of course, we can't do *exactly* circular, so there's always going to be some direction quantization, but we can do the best possible binary pixel approximation.)

Reply:
Yes, I think this may be worth doing. I'm not sure if I want to put it into this PR, but as a small issue to improve precision it sounds like a good idea.

Reply:
Issue created.